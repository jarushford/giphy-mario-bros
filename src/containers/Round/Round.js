import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import '../../main.scss'

export function Round({ round, players }) {
  if (!players.length) {
    return <div />
  }

  return (
    <div className="round">
      <h3 className="round-num">{round + 1}</h3>
      <h3 className="round-sep">/</h3>
      <h3 className="round-total">{players.length * 2 + 1}</h3>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  round: state.round,
  players: state.players
})

export default connect(mapStateToProps)(Round)

Round.propTypes = {
  round: PropTypes.number.isRequired,
  players: PropTypes.array.isRequired
}