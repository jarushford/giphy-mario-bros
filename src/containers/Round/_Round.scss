.round {
  position: absolute;
  top: 0px;
  left: 20px;
  display: flex;
  color: rgb(233, 12, 12);
  text-shadow: $outlinethin;
  font-family: 'Mario';
  font-size: 2rem;
}

.round-num {
  position: relative;
  bottom: 10px;
}

.round-total {
  position: relative;
  top: 10px;
}