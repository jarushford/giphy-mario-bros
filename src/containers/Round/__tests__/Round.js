import React from 'react'
import { Round, mapStateToProps } from '../Round'
import { shallow } from 'enzyme'

describe('Round', () => {
  it('should match the snapshot', () => {
    const mockPlayers = ['P1', 'P2', 'P3']
    const wrapper = shallow(<Round round={1} players={mockPlayers} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with no players', () => {
    const mockPlayers = []
    const wrapper = shallow(<Round round={1} players={mockPlayers} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return a props object with round and players', () => {
      const mockState = {
        round: 2,
        players: ['P1', 'P2', 'P3'],
        judge: 1,
        currentPlayer: 2
      }
      const expected = {
        round: 2,
        players: ['P1', 'P2', 'P3']
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})