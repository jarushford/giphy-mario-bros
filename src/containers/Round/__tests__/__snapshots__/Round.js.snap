// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Round should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Round
    players={
      Array [
        "P1",
        "P2",
        "P3",
      ]
    }
    round={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="round-num"
        >
          2
        </h3>,
        <h3
          className="round-sep"
        >
          /
        </h3>,
        <h3
          className="round-total"
        >
          7
        </h3>,
      ],
      "className": "round",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 2,
          "className": "round-num",
        },
        "ref": null,
        "rendered": 2,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "/",
          "className": "round-sep",
        },
        "ref": null,
        "rendered": "/",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 7,
          "className": "round-total",
        },
        "ref": null,
        "rendered": 7,
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="round-num"
          >
            2
          </h3>,
          <h3
            className="round-sep"
          >
            /
          </h3>,
          <h3
            className="round-total"
          >
            7
          </h3>,
        ],
        "className": "round",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "round-num",
          },
          "ref": null,
          "rendered": 2,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "/",
            "className": "round-sep",
          },
          "ref": null,
          "rendered": "/",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 7,
            "className": "round-total",
          },
          "ref": null,
          "rendered": 7,
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Round should match the snapshot with no players 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Round
    players={Array []}
    round={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
